packages/salesforcedx-vscode-core/src/commands/forceSourceToolingDeploy.ts

```diff
@@ -0,0 +1,162 @@
+ */
+
+import {
+  Command,
+  CompositeCliCommandExecution,
+  CompositeCliCommandExecutor,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  Row,
+  Table
+} from '@salesforce/salesforcedx-utils-vscode/out/src/output';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { ToolingDeploy, ToolingRetrieveResult } from '../deploys';
+import { nls } from '../messages';
+import { OrgAuthInfo, ToolingDeployParser } from '../util';
+import { SfdxCommandletExecutor } from './util';
+
+export class DeployRetrieveExecutor extends SfdxCommandletExecutor<{}> {
+  public build(sourcePath: string): Command {
+    const commandBuilder = new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_source_deploy_text'))
+      .withArg('force:source:deploy')
+      .withLogName('force_source_tooling_deploy')
+      .withFlag('--sourcepath', sourcePath)
+      .withJson();
+    return commandBuilder.build();
+  }
+
+  public async execute(response: ContinueResponse<string>): Promise<void> {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const executionWrapper = new CompositeCliCommandExecutor(
+      this.build(response.data)
+    ).execute(cancellationToken);
+    this.attachExecution(
+      executionWrapper,
+      cancellationTokenSource,
+      cancellationToken
+    );
+    executionWrapper.processExitSubject.subscribe(() => {
+      this.logMetric(executionWrapper.command.logName, startTime);
+    });
+
+    try {
+      const deployLibrary = new ToolingDeploy();
+      const usernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias(true);
+      let username: string | undefined;
+      if (usernameOrAlias) {
+        username = await OrgAuthInfo.getUsername(usernameOrAlias);
+      }
+      await deployLibrary.init(username!);
+      const deployOutput = await deployLibrary.deploy({
+        FilePathOpts: { filepath: response.data }
+      });
+
+      const parser = new ToolingDeployParser(deployOutput!);
+      await this.outputResult(executionWrapper, parser);
+    } catch (e) {
+      const deployOutput = {
+        State: 'Error',
+        ErrorMsg: e.message
+      } as ToolingRetrieveResult;
+      const parser = new ToolingDeployParser(deployOutput);
+      await this.outputResult(executionWrapper, parser, response.data);
+    }
+  }
+
+  public async outputResult(
+    executionWrapper: CompositeCliCommandExecution,
+    parser: ToolingDeployParser,
+    sourceUri?: string
+  ) {
+    const table = new Table();
+    let title: string;
+    switch (parser.result.State) {
+      case 'Completed':
+        title = nls.localize(`table_title_deployed_source`);
+        const successRows = parser.buildSuccesses(
```

Comment:
The first time I ran a deploy I got an issue from somewhere in here (Row is missing the key filePath) . I then ran it again without making any change to the file being deployed and it was successful.

![Screen Shot 2020-03-04 at 2 25 26 PM](https://user-images.githubusercontent.com/1041842/75928877-3472e500-5e24-11ea-9cf9-5e173ec193ac.png)